src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) Eficent Business and IT Consulting Services S.L. 2019
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+import random
+import subprocess
+
+import requests
+from bs4 import BeautifulSoup
+
+from .. import github
+from ..config import REBASE_BOT_INTRO_MESSAGES, switchable
+from ..manifest import git_modified_addon_dirs, is_maintainer
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+def _git_call(cmd, cwd="."):
+    subprocess.check_output(
+        cmd, universal_newlines=True, stderr=subprocess.STDOUT, cwd=cwd
+    )
+
+
+def _get_rebase_bot_intro_message():
+    i = random.randint(0, len(REBASE_BOT_INTRO_MESSAGES) - 1)
+    return REBASE_BOT_INTRO_MESSAGES[i]
+
+
+def _user_can_rebase(gh, org, repo, username, addons_dir, target_branch):
+    """
+    Check if a user is allowed to rebase.
+
+    addons_dir must be a git clone of the branch to merge to target_branch.
+    """
+    gh_repo = gh.repository(org, repo)
+    if github.github_user_can_push(gh_repo, username):
+        return True
+    modified_addon_dirs, other_changes = git_modified_addon_dirs(
+        addons_dir, target_branch
+    )
+    if other_changes or not modified_addon_dirs:
+        return False
+    # if we are modifying addons only, then the user must be maintainer of
+    # all of them on the target branch
+    current_branch = github.git_get_current_branch(cwd=addons_dir)
+    try:
+        _git_call(["git", "checkout", target_branch], cwd=addons_dir)
+        return is_maintainer(username, modified_addon_dirs)
+    finally:
+        _git_call(["git", "checkout", current_branch], cwd=addons_dir)
```

Comment:
I think this method and associated exception and comment on the PR should be factored out and generalized.
It's the same conditions for rebase or merge.